Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -55,6 +52,8 @@ namespace Terrain
                 ->Version(0)
                 ;
         }
+
+        TerrainDetailTextureComputePassData::Reflect(context);
```

Comment:
This feels like an awkward place to do the reflection to me. I think TerrainSystemComponent would fit better.

Reply:
The pass system is more close to the renderer. Since Mike and Galib also suggest putting it in the feature processor, I'll stick to it for now.